src/main/java/com/google/api/codegen/metacode/InitCodeContext.java

```diff
@@ -31,6 +31,10 @@
   public enum InitCodeOutputType {
     SingleObject,
     FieldList,
+    // This is used in grpc testing for when a method accepts a field list but the grpc request
+    // object is needed. This helps to compare the the request object that was constructed by the
+    // method against the request object that is initialized.
+    FieldListAndRequestObject
```

Comment:
Why not use a mocking library like the other languages do? Then you can easily get out the passed in argument after running the method and use `test.asserts` to check the request fields.